pkg/tokenvalidation/validationconfig.go

```diff
@@ -0,0 +1,170 @@
+		kubeAPIServerClientConfig: c.GenericConfig.ClientConfig,
+
+		TokenValidationConfig: *c.TokenValidationConfig,
+	}
+
+	return cfg
+}
+
+// New returns a new instance of TokenValidationServer from the given config.
+func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget) (*TokenValidationServer, error) {
+	genericServer, err := c.GenericConfig.New("oauth.openshift.io-tokenvalidation", delegationTarget)
+	if err != nil {
+		return nil, err
+	}
+
+	s := &TokenValidationServer{
+		GenericAPIServer:                    genericServer,
+		AccessTokenInactivityTimeoutSeconds: c.TokenValidationConfig.AccessTokenInactivityTimeoutSeconds,
+	}
+
+	s.installTokenValidation(c.kubeAPIServerClientConfig)
+
+	return s, nil
+}
+
+func (s *TokenValidationServer) installTokenValidation(kubeAPIServerClientConfig *restclient.Config) error {
+	// build clients
+	userClient, err := userclients.NewForConfig(s.GenericAPIServer.LoopbackClientConfig)
+	if err != nil {
+		return err
+	}
+
+	oauthClient, err := oauthclients.NewForConfig(s.GenericAPIServer.LoopbackClientConfig)
+	if err != nil {
+		return err
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(kubeAPIServerClientConfig)
+	if err != nil {
+		return err
+	}
+	bootstrapUserDataGetter := bootstrap.NewBootstrapUserDataGetter(kubeClient.CoreV1(), kubeClient.CoreV1())
+
+	// create informer for the users to be used in user <-> groups mapping
+	userInformer := userinformer.NewSharedInformerFactory(userClient, defaultInformerResyncPeriod)
+	if err := userInformer.User().V1().Groups().Informer().AddIndexers(cache.Indexers{
+		usercache.ByUserIndexName: usercache.ByUserIndexKeys,
+	}); err != nil {
+		return err
+	}
+
+	s.GenericAPIServer.AddPostStartHookOrDie("openshift.io-StartUserInformer", func(ctx genericapiserver.PostStartHookContext) error {
+		go userInformer.Start(ctx.StopCh)
+		return nil
+	})
+
+	// add validators
+	validators := []tokenvalidators.OAuthTokenValidator{tokenvalidators.NewExpirationValidator(), tokenvalidators.NewUIDValidator()}
+	groupMapper := usercache.NewGroupCache(userInformer.User().V1().Groups())
+	if s.AccessTokenInactivityTimeoutSeconds != 0 {
+		oauthInformer := oauthinformer.NewSharedInformerFactory(oauthClient, defaultInformerResyncPeriod)
+
+		timeoutValidator := tokenvalidators.NewTimeoutValidator(
+			oauthClient.OauthV1().OAuthAccessTokens(),
+			oauthInformer.Oauth().V1().OAuthClients().Lister(),
+			s.AccessTokenInactivityTimeoutSeconds,
+			minimumInactivityTimeoutSeconds)
+
+		validators = append(validators, timeoutValidator)
+
+		s.GenericAPIServer.AddPostStartHookOrDie("openshift.io-StartOAuthInformer", func(ctx genericapiserver.PostStartHookContext) error {
+			go oauthInformer.Start(ctx.StopCh)
+			return nil
+		})
+		s.GenericAPIServer.AddPostStartHookOrDie("openshift.io-StartTokenTimeoutUpdater", func(ctx genericapiserver.PostStartHookContext) error {
+			go oauthInformer.Start(ctx.StopCh)
```

Comment:
is this intentional or you meant to start timeout validator?

Reply:
that was a mistake, it's fixed now